lib/arc_ecto/schema.ex

```diff
@@ -53,4 +22,37 @@ defmodule Arc.Ecto.Schema do
 
     end) || params
   end
+
+  # If given a changeset, apply the changes to obtain the underlying data
+  # It would be nice to know under what context this will not be a changeset.
+  defp normalize_scope changeset_or_data do
+    case changeset_or_data do
+      %Ecto.Changeset{} -> Ecto.Changeset.apply_changes(changeset_or_data)
+      %{__meta__: _} -> changeset_or_data #TODO: wont this cause normalize_params to fail?
```

Comment:
You can pass in a struct or a changeset to any cast function, and it would be wise to follow that same pattern for this `cast_attachments` function.

Let's say you want a separate workflow for updating an avatar: 

```elixir
defmodule User do
  def update_avatar(data, params) do
    data
    |> cast_attachments(params, [:avatar])
  end
end

# From controller:
Repo.one!(User) |> User.update_avatar(params) |> Repo.update!()
```